src/records.rs

```diff
@@ -151,16 +158,22 @@ const MAGIC_BYTE_OFFSET: usize = 16;
 impl RecordBatchEncoder {
     /// Encode records into given buffer, using provided encoding options that select the encoding
     /// strategy based on version.
```

Comment:
Please add documentation to all of the encode/decode methods that a None compressor will result in the correct compressor being automatically applied.